assets/src/modules/SingleWMSLayer.js

```diff
@@ -0,0 +1,331 @@
+        this._layer = null;
+
+        /**
+         * timeout function to manage the image layer reload
+         * @type {?function} 
+         */
+        this._timeout = null
+
+        /**
+         * the WMS Ratio.
+         * @type {number} 
+         */
+        this._WMSRatio = 1.1;
+
+        /**
+         * the image format
+         * @type {string}
+         * @todo could the format be readed from project config too?
+         */
+        this._format = "image/png";
+
+        /**
+         * minimun map scale
+         * @type {number}
+         */
+        this._minScale = 1;
+
+        /**
+         * maximum map scale
+         * @type {number}
+         */
+        this._maxScale = 1000000000000;
+
+        /**
+         * meters per units
+         * @type {number}
+         */
+        this._metersPerUnit = mainMapInstance.getView().getProjection().getMetersPerUnit();
+
+        /**
+         * ordered layers
+         * @type {String[]}
+         */
+        this._orderedLayers = [];
+
+        // construct base and map layers array
+        this._singleWMSLayerList.forEach((m,k) => {
+            if (m instanceof BaseLayerState) {
+                this._baseLayers.push(k);
+            } else if (m instanceof MapLayerState){
+                this._mapLayers.push(k);
+            }
+        });
+
+        this._orderedLayers = this._baseLayers.concat(this._mapLayers);
+        // initialize single Image layer
+        this.initializeLayer();
+
+        // register listener for map layers
+        mainLizmap.state.rootMapGroup.addListener(
+            evt => {
+                if (this._mapLayers.includes(evt.name)) {
+                    // the laeyr is included in the single WMS request
+
+                    // wait a bit in order to reduce the amount of requests 
+                    // e.g. when user turn on/off layers quickly
+                    clearTimeout(this._timeout);
+                    this._timeout = setTimeout(()=>{
+                        this.updateMap();
+                    },600)
+                }
+            },
+            ['layer.visibility.changed','group.visibility.changed','layer.symbol.checked.changed', 'layer.style.changed', 'layer.selection.token.changed', 'layer.filter.token.changed']
+        );
+
+        // register listener for base layers
+        mainLizmap.state.baseLayers.addListener(
+            evt => {
+                this.updateMap();
+            },
+            ['baselayers.selection.changed']
+        );
+
+    }
+    /**
+     * @type {ImageLayer<ImageWMS>}
+     * 
+     */
+    get layer(){
+        return this._layer;
+    }
+    /**
+     * creates the layer instance and get the startup single image 
+     * @memberof SingleWMSLayer
+     */
+    initializeLayer(){
+        let minResolution =  Utils.getResolutionFromScale(this._minScale, this._metersPerUnit);
+        let maxResolution =  Utils.getResolutionFromScale(this._maxScale, this._metersPerUnit); 
+
```

Comment:
You need to take into account the `wmsMaxWidth` and `wmsMaxHeight` https://github.com/3liz/lizmap-web-client/blob/master/assets/src/modules/map.js#L115 and the `pixelRatio` https://github.com/3liz/lizmap-web-client/blob/master/assets/src/modules/map.js#L121

Reply:
Agree. In addition, based on these info, if `useTileWms` is true, then I should load the whole wms image as `TileWMS`

Reply:
@rldhont 
for single layers `minResolution` and `maxResolution` are calculated for each node. Since with singleWMS these informations are lost, is it ok to left these parameters `undefined` for the singleWMS?